vernac/vernacentries.ml

```diff
@@ -1586,19 +1586,18 @@ let vernac_check_may_eval ~atts redexp glopt rc =
   let glopt = query_command_selector ?loc:atts.loc glopt in
   let (sigma, env) = get_current_context_of_args glopt in
   let sigma', c = interp_open_constr env sigma rc in
-  let c = EConstr.Unsafe.to_constr c in
   let sigma' = Evarconv.solve_unif_constraints_with_heuristics env sigma' in
   Evarconv.check_problems_are_solved env sigma';
-  let sigma',nf = Evarutil.nf_evars_and_universes sigma' in
+  let sigma',_ = Evarutil.nf_evars_and_universes sigma' in
   let uctx = Evd.universe_context_set sigma' in
   let env = Environ.push_context_set uctx (Evarutil.nf_env_evar sigma' env) in
-  let c = nf c in
+  let c_ground = EConstr.to_constr sigma' c in
   let j =
-    if Evarutil.has_undefined_evars sigma' (EConstr.of_constr c) then
-      Evarutil.j_nf_evar sigma' (Retyping.get_judgment_of env sigma' (EConstr.of_constr c))
+    if Evarutil.has_undefined_evars sigma' c then
+      Evarutil.j_nf_evar sigma' (Retyping.get_judgment_of env sigma' c)
```

Comment:
Why not the normalized c?

Reply:
Not sure what you mean, but I think `c_ground` should disappear at some point.

Reply:
Once `Arguments_renaming` is fixed.

Reply:
I just feel like we're normalizing multiple times for no reason.
I guess `Eval` and company are not performance critical so it's no a huge deal.

Reply:
Ok I'll do the tweak and add a comment, anyways this PR need a bench.